content/general/functions.textile

```diff
@@ -8,17 +8,176 @@ Reactor Functions allow your server-less functions to be invoked following chann
+
+When a message is published on a channel, a Reactor Function message event is triggered, which in turn will send a message to your server-less function.
+
+Every request will contain your message within the @body@ part of the request. The contents of the @body@ depends on the format of your messages, and whether you've used an envelope or not.
+
+* No Envelope - Without an envelope, the contents of @body@ will simply be the contents message that was sent on the Ably Channel. 
+
+* Envelope - If you use an envelope, the @body@ will contain a bunch of extra attributes, which are the following:
+
+
+- source := the source of the request, namely "@channel.message"
+- appId := the ID of your application with Ably
+- channel := name of the channel the Ably Message was published on
+- site := an internal site identifier indicating the data center from which the message was published
+- ruleId := an ID for the rule which has created this request
+- messages := an @Array@ of messages with fields for each message described below
+
+Each member of @messages@ has the following fields:
+
+- id := unique ID assigned by Ably to this message
+- name := a string representing the event name for the published message, see the "publish method":/rest/
+- connectionId := the public unique identifier for the publisher's connection. Find out more about "@connectionId@":/realtime/connection#id
+- timestamp := the time in milliseconds since the epoch when this message was received by Ably
+- data := the data payload for the published message. Find out more about "message data":/rest/channels-messages#message
+
+h4. Decoding messages
+
+Each member of @messages@ is a 'raw' message. Messages sent "over the realtime service":/realtime/channels-messages are automatically decoded into "@Messages@":/realtime/types#message object by the Ably client library. With Reactor Functions you need to to do this explicitly, using "@Message.fromEncodedArray@":/realtime/channels-messages#message-from-encoded-array on the @messages@ array, or "@Message.fromEncoded@":/realtime/channels-messages#message-from-encoded on an individual member of that array. This will transform them into an array of "@Messages@":/realtime/types#message (or in the case of @fromEncoded@, an individual "@Message@":/realtime/types#message). This has several advantages, e.g.:
+
+* It will fully decode any @data@ (using the @encoding@) back into the same datatype that it was sent in (or an equivalent in each client library's language)
+* If you are using "encryption":/realtime/encryption, you can pass your encryption key to the method and it will decrypt the @data@ for you
+
+We recommend you do this for all messages you receive through Reactor Functions. For example (using ably-js):
+
+```[javascript]
+  const messages = Ably.Realtime.Message.fromEncodedArray(request.messages);
```

Comment:
Why `request.messages`, given the payload below, it should be `request.body.items` assuming we are using a Javascript library that exposes a body from a request object (Lambda, Express, and many more)